src/core/Makefile.am

```diff
@@ -100,6 +100,7 @@ libxlio_la_SOURCES := \
 	event/delta_timer.cpp \
 	event/event_handler_manager.cpp \
 	event/event_handler_manager_local.cpp \
+	event/group.cpp \
```

Comment:
I suggest to rename it to xlio_ring_group. Both the class and the file. Otherwise it suggests some general group code. But in fact this is a specific class for ring group

Reply:
I wouldn't say this is a ring group. It's more socket group from user perspective and the ring collection is just implementation specific.
The idea of the group is to create a serialization entity or a context with a unique set of sockets. Operations with different groups can be executed in parallel and user needs to care about serialization only within a group.
Also, ring is an internal abstraction which shouldn't be exposed to user. Since we expose the group abstraction to user, I'd like to avoid mentioning the ring. If you think `xlio_group` is too generic, probably `xlio_socket_group` will be better?

Reply:
xlio_sockets_group sounds right!

Reply:
will rename to `xlio_polling_group`

Reply:
Done.